fs/span-manager/span_manager.go

```diff
@@ -88,12 +89,23 @@ type spanInfo struct {
 
 // New creates a SpanManager with given ztoc and content reader, and builds all
 // spans based on the ztoc.
+
+// Should this return errors/nil objects on failure?
 func New(ztoc *ztoc.Ztoc, r *io.SectionReader, cache cache.BlobCache, retries int, cacheOpt ...cache.Option) *SpanManager {
 	index, err := ztoc.Zinfo()
 	if err != nil {
 		return nil
 	}
 	ztoc.Checkpoints = nil
+
+	b := make([]byte, index.StartCompressedOffset(0))
```

Comment:
I think this could use a comment to explain that it’s also the only place we use the layer gzip/compression header and that this has a second purpose of ensuring we eventually read every byte from the layer.